pkg/controller/persistence/dgraph/pod.go

```diff
@@ -14,38 +38,128 @@ type ID struct {
+	IsPod      bool         `json:"isPod,omitempty"`
 	Name       string       `json:"name,omitempty"`
 	StartTime  time.Time    `json:"startTime,omitempty"`
 	EndTime    time.Time    `json:"endTime,omitempty"`
 	Containers []*Container `json:"containers,omitempty"`
+	Interacts  []*Pod       `json:interacts,omitempty`
 }
 
 type Container struct {
 	ID
-	IsContainer string    `json:"isContainer"`
+	IsContainer bool      `json:"isContainer,omitempty"`
 	Name        string    `json:"name,omitempty"`
 	StartTime   time.Time `json:"startTime,omitempty"`
 	EndTime     time.Time `json:"endTime,omitempty"`
 	Pod         Pod       `json:"pod,omitempty"`
 }
 
+func CreatePod(pod api_v1.Pod, xid string) (*api.Assigned, error) {
+	newPod := Pod{
+		Name:  pod.Name,
+		IsPod: true,
+		ID:    ID{Xid: xid},
+	}
+	bytes, err := json.Marshal(newPod)
+	if err != nil {
+		return nil, err
+	}
+	return MutateNode(Client, bytes)
+}
+
 func PersistPod(pod api_v1.Pod) error {
 	xid := pod.Namespace + ":" + pod.Name
-	uid, _ := GetUId(Client, xid, "isPod")
+	uid, _ := GetUId(Client, xid, IsPod)
 
+	var newPod Pod
 	if uid == "" {
-		// If pod is not present, persist it.
-		newPod := Pod{
-			Name:  pod.Name,
-			IsPod: "",
-			ID:    ID{Xid: xid},
+		assigned, err := CreatePod(pod, xid)
+		if err != nil {
+			return err
+		}
+		uid = assigned.Uids["blank-0"]
+		newPod = Pod{
+			ID:         ID{Xid: xid, UID: uid},
+			Containers: GetContainers(pod, uid),
+		}
+	} else {
+		newPod = Pod{
+			ID:         ID{Xid: xid, UID: uid},
+			Containers: GetContainers(pod, uid),
+		}
+	}
+	bytes, err := json.Marshal(newPod)
+	if err != nil {
+		return err
+	}
+	_, err = MutateNode(Client, bytes)
+	return err
+}
+
+func GetContainers(pod api_v1.Pod, podUid string) []*Container {
+	podXid := pod.Namespace + ":" + pod.Name
+
+	containers := []*Container{}
+	for _, c := range pod.Spec.Containers {
+		containerXid := podXid + ":" + c.Name
+		uid, _ := GetUId(Client, containerXid, IsContainer)
+
+		var container *Container
+		if uid == "" {
+			container = &Container{
+				ID:          ID{Xid: containerXid, UID: uid},
+				Name:        c.Name,
+				IsContainer: true,
+				Pod:         Pod{ID: ID{Xid: podXid, UID: podUid}},
+			}
+		} else {
+			container = &Container{
+				ID:  ID{Xid: containerXid, UID: uid},
+				Pod: Pod{ID: ID{Xid: podXid, UID: podUid}},
+			}
 		}
-		bytes, err := json.Marshal(newPod)
+
+		containers = append(containers, container)
+	}
+	return containers
+}
+
+func PersistPodsInteractionGraph(sourcePod string, destinationPods []string) error {
+	uid, err := GetUId(Client, sourcePod, IsPod)
+	if err != nil {
+		return err
+	}
+	if uid == "" {
+		log.Println("Source Pod " + sourcePod + " is not persisted yet.")
+	}
+
+	pods := []*Pod{}
+	for _, destinationPod := range destinationPods {
+		uid, err := GetUId(Client, destinationPod, IsPod)
 		if err != nil {
 			return err
 		}
-		return MutateNode(Client, bytes)
+
+		if uid == "" {
```

Comment:
WIP if uid is ""?

Reply:
If pod is not already persisted then we don't update "interacts" edge.